sycl/doc/extensions/proposed/sycl_ext_oneapi_async_memory_alloc.asciidoc

```diff
@@ -0,0 +1,838 @@
+this extension is in line with the
+link:../experimental/sycl_ext_oneapi_enqueue_functions.asciidoc[
+  sycl_ext_oneapi_enqueue_functions] extension, so events are returned when
+calling `submit_with_event` and the `handler` overloads of these enqueue
+functions.
+
+[source,c++]
+----
+namespace ext::oneapi::experimental {
+
+void *async_malloc(const queue &que, usm::alloc kind, size_t size);
+
+void *async_malloc(handler &cgh, usm::alloc kind, size_t size);
+
+void *async_malloc_from_pool(const queue &que, size_t size,
+  const memory_pool &pool);
+
+void *async_malloc_from_pool(handler &cgh, size_t size,
+  const memory_pool &pool);
+
+void async_free(const queue &que, void *ptr);
+
+void async_free(handler &cgh, void *ptr);
+
+}  // ext::oneapi::experimental
+----
+
+[source, c++]
+----
+void *async_malloc(const queue &que, usm::alloc kind, size_t size);
+
+void *async_malloc(handler &cgh, usm::alloc kind, size_t size);
+
+void *async_malloc_from_pool(const queue &que, size_t size,
+  const smemory_pool &pool);
+
+void *async_malloc_from_pool(handler &cgh, size_t size,
+  const memory_pool &pool);
+----
+
+_Effects_: Enqueues a command to `q` or the SYCL queue associated with `h` which
+will asynchronously allocate memory of size `size` in bytes, allocating from the
+memory pool `pool` if provided, otherwise allocation from the default memory
+pool associated with the SYCL context and device associated with `q` or `h`.
+If a memory pool `pool` is provided the allocation kind is that of the memory
+pool, otherwise if an allocation kind `kind` parameter is provided, this is the
+allocation kind which is used. If a memory pool `pool` is provided, this must be
+associated with the same SYCL context and device as `q` or the SYCL queue
+associated with `h`. Memory can either be allocated directly from the memory
+pool or allocated to the memory pool to provide enough memory in the memory pool
+for the allocation. Accessing the memory at the address of the pointer returned
+by asynchronous malloc functions before the command has completed execution is
+undefined behavior.
+
+_Returns_: A pointer to the address of a memory reservation. When the allocation
+size is zero bytes, these functions behave in a manner consistent with C++
+`std::malloc`. The value returned is unspecified in this case, and the returned
+pointer may not be used to access storage. If this pointer is not `null`, it
+must be passed to `sycl::async_free` to avoid a memory leak.
+
+_Throws_: An exception with the `errc::memory_allocation` error code if the
+allocation brings the memory pool over the maximum size. This error must be
+thrown asynchronously. An exception with the `errc::invalid` error code if
+`kind` is `usm::alloc::unknown`.
+
+[source, c++]
+----
+void async_free(const queue &que, void *ptr);
+
+void async_free(handler &cgh, void *ptr);
+----
+
+_Effects_: Enqueues a command to `q` or the SYCL queue associated with `h` which
```

Comment:
```suggestion
_Effects_: Enqueues a command to `que` or the SYCL queue associated with `cgh` which
```

Reply:
Fixed in https://github.com/intel/llvm/pull/14800/commits/d154a67861e46477ba07b415b845348f880ce8a4.